src/openms/include/OpenMS/COMPARISON/CLUSTERING/GridBasedClustering.h

```diff
@@ -539,17 +539,23 @@ namespace OpenMS
     {
       int A1 = c1.getPropertyA();
       int A2 = c2.getPropertyA();
-      std::vector<int> B1 = c1.getPropertiesB();
-      std::vector<int> B2 = c2.getPropertiesB();
 
       // check if any of the properties A and B is not set i.e. =-1
-      if (A1 == -1 || A2 == -1 || std::find(B1.begin(), B1.end(), -1) != B1.end() || std::find(B2.begin(), B2.end(), -1) != B2.end())
+      if (A1 == -1 || A2 == -1)
       {
         return false;
       }
 
       // Will the merged cluster have the same properties A?
-      bool vetoA = !(A1 == A2);
+      if (A1 != A2) return true;
+
+      std::vector<int> B1 = c1.getPropertiesB();
+      std::vector<int> B2 = c2.getPropertiesB();
+
```

Comment:
Please add comment.

// Check if the properties B of both clusters have been set or not. (i.e. =-1 if not set)